### ISSUE TITLE
ACF block's saved data (repeater sub-fields) silently blanked on normal Update — block `mode` flips to "preview" even without touching preview/expanded-editor UI

### ISSUE THREAD

**Issue by bmohammadi-svg** [AdvancedCustomFields/acf#1026]
TITLE:
ACF block's saved data (repeater sub-fields) silently blanked on normal Update — block `mode` flips to "preview" even without touching preview/expanded-editor UI

BODY:
**Describe the bug**

A block registered via `acf_register_block_type()` (no `acf_block_version` set, so default V2/`mode: edit` behavior) containing a **repeater** field can have its `mode` attribute flip from `"edit"` to `"preview"` on a normal Save/Update click — with no one touching the block's preview/expanded-editor toggle. When this happens, every repeater sub-field value (e.g. `image`) is serialized as an **empty string** in the saved `post_content`, and the front end renders the block's static registration-time preview instead of the real content. This has happened twice in production for us in the last 24 hours, both times on ordinary edits (removing/updating repeater rows) via the standard sidebar "Block" panel — never via the pencil/expand icon or any preview toggle.

**To Reproduce**

We were not able to build a deterministic, isolated repro — this is what we reconstructed from two real production incidents via WordPress revision history and Apache access logs:

1. Register a block via `acf_register_block_type()` with `'mode' => 'edit'` and a repeater sub-field (e.g. image + text), no `acf_block_version` set.
2. Open a page containing the block in the block editor. Click the block (not the pencil/expand icon) and edit a repeater row in the standard sidebar "Block" panel — the panel shows correct, real data.
3. Click Update.
4. Compare the resulting revision to the one before it: in both of our incidents, every block on the page had `"mode":"edit"` flip to `"mode":"preview"`, and every repeater image sub-field flipped from a real attachment ID to `""`.
5. In both cases, saving again (Update) a second time, ~15–50 seconds later, produced a correct revision (`mode:"edit"`, real values restored) — so the underlying field data was never actually lost, just the specific save was corrupted.

Apache access logs around the bad save show a burst of `POST /wp-admin/admin-ajax.php` requests (consistent with `acf/ajax/fetch-block`) in the 10–30 seconds immediately before the `POST /wp-json/wp/v2/pages/{id}` save request that produced the bad revision — in both incidents. We were not able to capture the exact in-browser request bodies live (both incidents were caught after the fact from logs), but the timing strongly suggests a race between an in-flight block preview/fetch and the block's attributes being read for serialization at save time — similar in spirit to the `isFetchingBlock` timing issues described in #1025, but manifesting as silent data loss on a normal save rather than a stuck UI.

**Expected behavior**

Clicking Update should never serialize a block's in-progress/loading state (`mode: preview`, blank repeater values) over real, already-saved data — either by not allowing the save to proceed while a block's data is still resolving, or by never letting a block's committed attributes regress to a default/blank state once real values have loaded.

**Version Information**

- WordPress: 7.1
- Advanced Custom Fields PRO: 6.8.8
- PHP: 8.3.6
- Block registered via `acf_register_block_type()` (PHP array registration, not `block.json`), `'mode' => 'edit'`, no `acf_block_version` set.

**Additional context**

This may be related to two other things we found while investigating:

1. A separate, purely cosmetic issue: without `acf_block_version` set, the editor canvas shows the block's registered `example`/preview image instead of a live render whenever the block isn't the actively-selected block. Setting `acf_block_version: 3` fixes that, but on our repeater blocks it breaks the V3 "expanded editor" panel instead — every repeater sub-field renders blank in that panel's form (real data, correct on the server side, just not applied to the panel's inputs), which is why we reverted to the V2 default and are hitting the bug described above instead.
2. #1025 (filed today) describes the Expanded Editor becoming unresponsive during an in-flight block preview re-fetch (`isFetchingBlock`), which reads to us like the same underlying fetch/attribute-reconciliation timing area of the codebase, just a different symptom (stuck UI there, silently-blanked saved data here).

Happy to provide the exact revision content diffs (before/after JSON) and access log excerpts from our two incidents if useful — redacted to remove real customer data.

**Comment by bmohammadi-svg** [comment]
Reproduced this again today (2026-09-24) on the current release, ACF PRO 6.8.10, WordPress 7.1.2, PHP 8.5.4. Wanted to share fresh, direct database evidence since the original report was reconstructed after the fact from logs.

Setup: block registered via `acf_register_block_type()`, repeater field with image/headline/text/social sub-fields, `mode: edit`, no explicit `acf_block_version` set.

Steps:
1. Opened the page in the block editor, selected the block, opened the Block sidebar panel (not the pencil/expand icon).
2. Edited a repeater row's Text field in that panel.
3. Clicked Update right after typing, without waiting.

Result, confirmed by reading the row directly from the database rather than trusting the editor UI:

```
small_teaser_repeater_0_headline: ""
small_teaser_repeater_0_text: ""
small_teaser_repeater_0_image: ""
... (same for all 7 rows, every sub-field)
```

All 7 repeater rows across every sub-field were saved as empty strings in `post_content`. `mode` stayed `"edit"` this time (did not flip to `"preview"` like the original report), so that symptom may be a separate signal of the same underlying race rather than a required trigger. The block editor's own UI still showed the correct values after the save, since it was working from stale in-memory state, not the value that actually got written. The corruption was only visible by reading `post_content` from the database directly, or by reloading the editor.

Recovered immediately via `wp_restore_post_revision()` against the last good revision, so no lasting data loss, but this confirms the bug is still present on the current release and is not tied to an old ACF version.

Happy to share the exact page/block setup or try anything else that would help narrow this down.
